src/devices/Common/Iot/Device/Multiplexing/IOutputSegment.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Threading;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// Interface that abstracts multiplexing over a segment of outputs.
+    /// </summary>
+    public interface IOutputSegment : IDisposable
+    {
+        /// <summary>
+        /// Length of segment (number of outputs)
+        /// </summary>
+        int Length { get; }
+
+        /// <summary>
+        /// Segment values.
+        /// </summary>
+        PinValue this[int index] { get; }
+
+        /// <summary>
+        /// CancellationToken for segment.
+        /// </summary>
+        CancellationToken CancellationToken { get; }
+
+        /// <summary>
+        /// Writes a PinValue to a multiplexed output.
+        /// Does not perform a latch.
+        /// </summary>
+        void Write(int output, PinValue value);
+
+        /// <summary>
+        /// Writes an int (processed as one of more bytes) to a multiplexed output.
+        /// Written one byte at a time, left to right. Least significant bit will written to index 0.
+        /// Does not perform a latch.
+        /// </summary>
+        void Write(int value);
+
+        /// <summary>
+        /// Turns off all outputs.
+        /// Performs a latch.
+        /// </summary>
+        void Clear();
```

Comment:
Should this be a bit more descriptive like `TurnOffAllOutputs` or something? Clear (specially in a segment which technically is a collection) has the connotation in C# of removing things from the collection.